zhaquirks/xiaomi/aqara/plug_eu.py

```diff
@@ -346,3 +354,102 @@ class PlugMAEU01Alt3(PlugMAEU01):
     }
 
     replacement = PlugMAEU01.replacement
+
+
+class AqaraPowerOutageMemoryEnum(types.uint8_t, Enum):
+    """Power Outage Memory enum."""
+
+    On = 0x00
+    Previous = 0x01
+    Off = 0x02
+    Inverted = 0x03
+
+
+class PlugAEU001Cluster(XiaomiAqaraE1Cluster):
+    """Custom cluster for Aqara lumi plug AEU001."""
+
+    attributes = {
+        0x0200: ("button_lock", types.uint8_t, True),
+        0x0202: ("charging_protection", types.Bool, True),
+        0x0203: ("led_indicator", types.Bool, True),
+        0x0206: ("charging_limit", types.Single, True),
+        0x020B: ("overload_protection", types.Single, True),
+        0x0517: ("power_on_behavior", AqaraPowerOutageMemoryEnum, True),
+    }
+
+
+class PlugAEU001MeteringCluster(MeteringCluster):
+    """Custom cluster for Aqara lumi plug AEU001."""
+
+    def _update_attribute(self, attrid, value):
+        if attrid == self.CURRENT_SUMM_DELIVERED_ID:
+            current_value = self._attr_cache.get(attrid, 0)
+            if value < current_value:
+                _LOGGER.debug(
+                    "Ignoring attribute update for %s: new value %s is less than current value %s",
+                    attrid,
+                    value,
+                    current_value,
+                )
+                return
+        super()._update_attribute(attrid, value)
+
+
+(
+    QuirkBuilder("Aqara", "lumi.plug.aeu001")
+    .friendly_name(model="Wall Outlet H2 EU", manufacturer="Aqara")
+    .removes(TemperatureMeasurement.cluster_id)
+    .adds(DeviceTemperature)
+    .removes(OnOff.cluster_id, endpoint_id=2)
+    .replaces(BasicCluster)
+    .replaces(PlugAEU001MeteringCluster)
+    .replaces(ElectricalMeasurementCluster)
+    .replaces(PlugAEU001Cluster)
+    .replaces(AnalogInputCluster, endpoint_id=21)
+    .switch(
+        attribute_name="button_lock",
+        cluster_id=PlugAEU001Cluster.cluster_id,
+        force_inverted=True,
+        translation_key="button_lock",
+        fallback_name="Button Lock",
```

Comment:
Per HA spec, only the first letter should be capitalized, all other words should be lowercase (except for abbreviations). So, this should be "Buton lock", although we may want to use `child_lock` for the `translation_key` (and `fallback_name` if that is the same, as that's what we use for other devices).

Please also adjust the names of the other entities. (E.g. "Power on behavior", "LED indicator", ...) Thanks!

Reply:
Changed to `child_lock` and updated the other names as well. 👍